### PR TITLE
payalgo: Throttle pay command if failure is due to blockheight disagreement

### PR DESCRIPTION
**Pull request by ZmnSCPxj**
Fixes: #1136 

@hkjn , please see this!

### CODE REVIEW

**Comment by ZmnSCPxj** [comment]
Not sure of how to test this easily....

**Comment by hkjn** [comment]
Thanks!

ACK for intent fwiw (not very familiar with the code yet), should alleviate the #1136 looping behavior as a stop-gap.

File: lightningd/payalgo.c [hunk 1]
```diff
@@ -122,14 +123,49 @@ static void json_pay_failure(struct pay *pay,
 	command_fail_detailed(pay->cmd, r->errorcode, data, "%s", msg);
 }
 
+/* Determine if we should delay before retrying. Return a reason
+ * string, or NULL if we will not retry */
+static const char *delay_before_retry(const tal_t *ctx,
```
Comment (cdecker) [inline lightningd/payalgo.c]: `should_delay_retry` sounds a bit more intuitive imho

File: lightningd/payalgo.c [hunk 2]
```diff
@@ -122,14 +123,49 @@ static void json_pay_failure(struct pay *pay,
...
+	}
+
+	return NULL;
+}
+
 /* Start a payment attempt. */
 static bool json_pay_try(struct pay *pay);
 
+/* Used when delaying. */
+static void v_pay_try(struct pay *pay)
```
Comment (cdecker) [inline lightningd/payalgo.c]: What's the `v_` prefix stand for?

Comment (ZmnSCPxj) [reply lightningd/payalgo.c]: `void` return. Sorry, ran out of ideas...

**Comment by rustyrussell** [comment]
Leaving this unmerged so @cdecker can diagnose ackbot and see why it didn't trigger on my msg.

**Pull request merged**